lib/view_model/exchange/exchange_view_model.dart

```diff
@@ -776,16 +782,70 @@ abstract class ExchangeViewModelBase extends WalletChangeListenerViewModel with
 
   int get receiveMaxDigits => receiveCurrency.decimals;
 
-  bool isCanCreateTrade(Trade trade) {
+  Future<CreateTradeResult> isCanCreateTrade(Trade trade) async {
     if (trade.provider == ExchangeProviderDescription.thorChain) {
       final payoutAddress = trade.payoutAddress ?? '';
       final fromWalletAddress = trade.fromWalletAddress ?? '';
       final tapRootPattern = RegExp(P2trAddress.regex.pattern);
 
       if (tapRootPattern.hasMatch(payoutAddress) || tapRootPattern.hasMatch(fromWalletAddress)) {
-        return false;
+        return CreateTradeResult(
+          result: false,
+          errorMessage: S.current.thorchain_taproot_address_not_supported,
+        );
+      }
+
+      final currenciesToCheckPattern = RegExp('0x[0-9a-zA-Z]');
+      final inputAddress = trade.inputAddress ?? '';
```

Comment:
this checks for the address that is owned by the exchange provider, which the user will send the funds to, to make the exchange, which is not what we want to check

we wanna check the user sending address, which would be `trade.fromWalletAddress` and also check the receiving address which would be `trade.payoutAddress` like the if condition for taproot above

how did you verify that this is working? the only way I see this being verified is that you manually changed the `trade.inputAddress` which should be returned from the exchange provider API, by then you should already know that this is not correct